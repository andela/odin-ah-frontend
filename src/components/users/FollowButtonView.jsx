import React from 'react';
import PropTypes from 'prop-types';

const FollowButtonView = ({
  text, styles, loading, onClickHandler
}) => (
  <div className="js-button-wrapper" styles={styles} onClick={onClickHandler}>
    {!loading ? text : <i className="fas fa-sync fa-spin" />}
  </div>
);

FollowButtonView.propTypes = {
  text: PropTypes.string.isRequired,
  styles: PropTypes.object,
  loading: PropTypes.bool,
  onClickHandler: PropTypes.func
};

FollowButtonView.defaultProps = {
  loading: false,
  styles: {},
  text: 'Follow'
};

export default FollowButtonView;
