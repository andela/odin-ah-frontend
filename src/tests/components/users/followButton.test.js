import React from 'react';
import { shallow } from 'enzyme';

import FollowButtonView from '../../../components/users/FollowButtonView';

// User is not following
const followProps = {
  loading: false,
  text: 'Follow',
  onClickHandler: jest.fn(),
  styles: {}
};

// User is following
const unfollowProps = {
  loading: false,
  text: 'Unfollow',
  onClickHandler: jest.fn(),
  styles: {}
};

// Button is transitioning to new state
const loaderProps = {
  loading: true,
  text: '',
  onClickHandler: jest.fn(),
  styles: {}
};

it('renders without crashing', () => {
  const wrapper = shallow(<FollowButtonView />);
  expect(wrapper).toMatchSnapshot();
});

it('renders follow state', () => {
  const wrapper = shallow(<FollowButtonView {...followProps} />);
  expect(wrapper.exists()).toBe(true);
});

it('renders with loading state', () => {
  const wrapper = shallow(<FollowButtonView {...loaderProps} />);
  expect(wrapper.exists()).toBe(true);
});

it('renders unfollow state', () => {
  const wrapper = shallow(<FollowButtonView {...unfollowProps} />);
  expect(wrapper.exists()).toBe(true);
});

it('calls handler function on click events', () => {
  const wrapper = shallow(<FollowButtonView {...followProps} />);
  wrapper.find('.js-button-wrapper').simulate('click');
  expect(followProps.onClickHandler).toHaveBeenCalled();
});
