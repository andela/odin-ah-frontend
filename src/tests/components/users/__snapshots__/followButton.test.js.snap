// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="js-button-wrapper"
  styles={Object {}}
>
  Follow
</div>
`;
